packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/services/ResourceLoader.java

```diff
@@ -34,18 +45,34 @@
 @Service
 public class ResourceLoader implements IResourceLoader {
 
+    private static final MigrationData EMPTY_MIGRATION_DATA = new MigrationData("none", "0");
+
     private final Logger logger = LoggerFactory.getLogger(ResourceLoader.class);
 
+    private final List<IMigrationParticipant> migrationParticipants;
+
+    public ResourceLoader(List<IMigrationParticipant> migrationParticipants) {
+        this.migrationParticipants = migrationParticipants;
+    }
+
     @Override
     public Optional<Resource> toResource(ResourceSet resourceSet, String id, String name, String content) {
         Optional<Resource> optionalResource = Optional.empty();
 
+        var migrationData = getMigrationDataFromDocumentContent(content).orElse(EMPTY_MIGRATION_DATA);
+        var migrationExtendedMetaData = new MigrationExtendedMetaData(this.migrationParticipants, migrationData);
+        var migrationJsonProcessor = new MigrationJsonProcessor(this.migrationParticipants, migrationData);
+
+        HashMap<Object, Object> options = new HashMap<>();
+        options.put(JsonResource.OPTION_EXTENDED_META_DATA, migrationExtendedMetaData);
+        options.put(JsonResource.OPTION_JSON_RESSOURCE_PROCESSOR, migrationJsonProcessor);
```

Comment:
I'll have to check but there may be other locations now where this code is necessary. We should probably provide those options as a dedicated service.

Reply:
So this code should actually be also used in org.eclipse.sirius.web.application.document.services.JSONExternalResourceLoaderService#getResource where the adapter is needed